internal/service/maintenancewindow/resource_maintenance_window_test.go

```diff
@@ -40,15 +40,35 @@ func TestAccConfigRSMaintenanceWindow_basic(t *testing.T) {
 				),
 			},
 			{
-				Config: configBasic(orgID, projectName, dayOfWeekUpdated, hourOfDayUpdated),
+				Config: configBasic(orgID, projectName, dayOfWeek, hourOfDayUpdated),
 				Check: resource.ComposeTestCheckFunc(
 					checkExists(resourceName),
 					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
-					resource.TestCheckResourceAttr(resourceName, "day_of_week", cast.ToString(dayOfWeekUpdated)),
+					resource.TestCheckResourceAttr(resourceName, "day_of_week", cast.ToString(dayOfWeek)),
 					resource.TestCheckResourceAttr(resourceName, "hour_of_day", cast.ToString(hourOfDayUpdated)),
 					resource.TestCheckResourceAttr(resourceName, "number_of_deferrals", "0"),
 				),
 			},
+			{
+				Config: configBasic(orgID, projectName, dayOfWeekUpdated, hourOfDay),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
common check function, or helper functions like AddAttrChecks or AddAttrSetChecks could be used to reduce duplication, e.g.: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/doc_indentation/internal/service/searchindex/resource_search_index_test.go#L234

Reply:
if it's ok I'd leave it like that - I like to keep tests siloed without having to jump to other functions to understand what's being asserted and how. Verbosity / code duplication in tests should not be an issue (tl;dr: write DAMP tests and DRY code)

Few references:
* https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html
* https://thomascannon.me/guides/verbose-tests

Reply:
But thanks for the comment!

Reply:
you can define the checks at the beginning of the test and reuse in its steps.